clean up E0404 explanation (#69887)
rust-lang/rust: src/librustc_error_codes/error_codes/E0404.md

> Something which is not a trait was used in a trait position, such as a bound

Reviewer: ```suggestion
A type that is not a trait was used in a trait position, such as a bound
```

fn bar<T: Foo>(t: T) {} // error: `Foo` is not a trait
```

> Please verify that the trait's name was not misspelled or othrewise that the

Reviewer: ```suggestion
Please verify that the trait's name was not misspelled or that the 
```
